xgboost_ray/tests/test_fault_tolerance.py

```diff
@@ -87,25 +113,98 @@ def setUp(self):
         if os.path.exists(self.die_lock_file):
             os.remove(self.die_lock_file)
 
-        ray.init(num_cpus=2, num_gpus=0)
+        ray.init(num_cpus=2, num_gpus=0, log_to_driver=True)
 
     def tearDown(self) -> None:
         if os.path.exists(self.tmpdir):
             shutil.rmtree(self.tmpdir)
         ray.shutdown()
 
-    def testTrainingContinuation(self):
+    def testTrainingContinuationKilled(self):
         """This should continue after one actor died."""
+        actors = [None, None]
+        additional_results = {}
+
+        bst = train(
+            self.params,
+            RayDMatrix(self.x, self.y),
+            callbacks=[_kill_callback(self.die_lock_file)],
+            num_boost_round=20,
+            ray_params=RayParams(max_actor_restarts=1, num_actors=2),
+            additional_results=additional_results,
+            _actors=actors)
+
+        x_mat = xgb.DMatrix(self.x)
+        pred_y = bst.predict(x_mat)
+        self.assertSequenceEqual(list(self.y), list(pred_y))
+        print(f"Got correct predictions: {pred_y}")
+
+        # End with two working actors
+        self.assertTrue(actors[0])
+        self.assertTrue(actors[1])
+
+        # Two workers finished, so N=32
+        self.assertEqual(additional_results["total_n"], 32)
+
+    def testTrainingContinuationElasticKilled(self):
+        """This should continue after one actor died."""
+        actors = [None, None]
+        additional_results = {}
+
+        bst = train(
+            self.params,
+            RayDMatrix(self.x, self.y),
+            callbacks=[_kill_callback(self.die_lock_file)],
+            num_boost_round=20,
+            ray_params=RayParams(
+                max_actor_restarts=1,
+                num_actors=2,
+                elastic_training=True,
+                max_failed_actors=1),
```

Comment:
This should raise an error if `max_failed_actors` is set to 0 right? Is it worth adding another test for that case?

Reply:
Added `testTrainingStopElastic`